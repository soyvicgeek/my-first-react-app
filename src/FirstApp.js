import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ item, subtitle }) => {

  return (
    <>
      <h1> { item } </h1>
      {/* <h1>{ JSON.stringify(hi) }</h1> */}
      <p> { subtitle }</p>
    </>
  );

}

FirstApp.propTypes = {
  item: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  subtitle: 'Default Subtitle'
}

export default FirstApp;