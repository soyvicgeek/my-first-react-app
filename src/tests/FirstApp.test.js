import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

//import { render } from '@testing-library/react';
import FirstApp from './../FirstApp';

describe('Pruebas en <FirstApp />', () => {

  /*==========================================
  Tests with Jest
  ==========================================*/
  /*test('Debe de mostrar el mensaje "Hola, Soy Goku"', () => {
    const saludo = 'Hola, Soy Goku';
    const { getByText } = render( <FirstApp item={ saludo }/>);
    expect( getByText( saludo ) ).toBeInTheDocument();
  })*/

  test('Debe de mostrar <PrimeraApp /> correctamente', () => {

    const item = 'Hola, Soy Goku';
    const wrapper = shallow( <FirstApp item={item}/>);

    expect( wrapper ).toMatchSnapshot();

  });

  test('Debe de mostrar el subtitulo enviado por props', () => {

    const item = 'Hola, Soy Goku';
    const subtitle = 'Default Subtitle';
    const wrapper = shallow(
      <FirstApp
        item={ item }
        subtitle={ subtitle }
      />
    );

    const textParrafo = wrapper.find('p').text().trim();
    expect(textParrafo).toBe( subtitle );
    //console.log(textParrafo);

  });

})