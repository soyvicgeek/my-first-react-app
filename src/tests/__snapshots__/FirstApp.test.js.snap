// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, Soy Goku
     
  </h1>
  <p>
     
    Default Subtitle
  </p>
</Fragment>
`;
